testsuite/testsuite-keycloak/src/test/java/org/wildfly/swarm/keycloak/KeycloakArquillianTest.java

```diff
@@ -15,40 +15,122 @@
  */
 package org.wildfly.swarm.keycloak;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.net.URL;
+import java.util.UUID;
+
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Form;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+
 import org.jboss.arquillian.container.test.api.Deployment;
-import org.jboss.arquillian.container.test.api.RunAsClient;
 import org.jboss.arquillian.junit.Arquillian;
 import org.jboss.shrinkwrap.api.Archive;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
-import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.junit.Assert;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.wildfly.swarm.Swarm;
 import org.wildfly.swarm.arquillian.CreateSwarm;
-import org.wildfly.swarm.spi.api.JARArchive;
+import org.wildfly.swarm.jaxrs.JAXRSArchive;
 
 /**
  * @author Bob McWhirter
  */
 @RunWith(Arquillian.class)
 public class KeycloakArquillianTest {
 
-    @Deployment(testable = false)
-    public static Archive createDeployment() {
-        JARArchive deployment = ShrinkWrap.create(JARArchive.class);
-        deployment.add(EmptyAsset.INSTANCE, "nothing");
+    @Deployment
+    public static Archive<?> createDeployment() throws Exception {
+        JAXRSArchive deployment = ShrinkWrap.create(JAXRSArchive.class);
+        deployment.addResource(SecuredApplication.class);
+        deployment.addResource(SecuredResource.class);
+        deployment.addAsResource("wildfly-swarm-keycloak-example-realm.json");
+        deployment.addAsResource("keycloak.json");
         return deployment;
     }
 
     @CreateSwarm
     public static Swarm newContainer() throws Exception {
```

Comment:
Is there a way to do this without `@CreateSwarm`?

Reply:
I don't know if it is possible to avoid creating Swarm explicitly :-) given that it has to be aware of the custom configuration resource and a  '-s' command line argument as opposed to a system property is used to let Swarm know about its location

Reply:
Adding a `project-defaults.yml` inside the Archive should mean it's picked up, so it should be possible to add into `createDeployment()`

Reply:
But the war name is dynamic

Reply:
It doesn't work if you call `Shrinkwrap.create(JAXRSArchive.class, "myName.war")`?

Reply:
I can definitely use Secured instead of generating a custom project config ? It is much simpler (thought I guess testing the config approach is preferred)

Reply:
Let me double check again :-), I don't think it works...

Reply:
No, does not work, I spent quite a lot of time with it last week, FYI, after specifying 'example-keycloak' as a war name, I get

"target/wfswarmexample-keycloak233151353883937125.war/WEB-INF/"

Reply:
ok, ignore me then